Monika After Story/game/script-ch30.rpy

```diff
@@ -1170,6 +1170,24 @@ label ch30_end:
 
 # label for things that may reset after a certain amount of time/conditions
 label ch30_reset:
+
+    # setup the docking station to handle the detection
+    $ rpyCheckStation = store.MASDockingStation(renpy.config.gamedir)
+
+    $ listRpy = rpyCheckStation.getPackageList(".rpy")
+
+    if len(listRpy) == 0 or persistent.current_monikatopic == "monika_rpy_files":
+        if len(listRpy) == 0 and persistent.current_monikatopic == "monika_rpy_files":
+            $ persistent.current_monikatopic = 0
+
+        if "monika_rpy_files" in persistent.event_list:
```

Comment:
since its possible to queue an event more than once, this should be a loop removal:
```suggestion
        while "monika_rpy_files" in persistent.event_list:
```